Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -247,6 +77,64 @@ private extension PrivateSwiftUIStatePropertyRule {
             violations.append(decl.bindingSpecifier.positionAfterSkippingLeadingTrivia)
         }
     }
+
+    final class Rewriter: ViolationsSyntaxRewriter {
+        /// LIFO stack that stores type inheritance clauses for each visited node
+        /// The last value is the inheritance clause for the most recently visited node
+        /// A nil value indicates that the node does not provide any inheritance clause
+        private var visitedTypeInheritances = Stack<InheritanceClauseSyntax?>()
```

Comment:
This is some duplication. There is a mechanism that allows you to collect replacements and additions in the visitor already, sparing you to implement a rewriter (and duplicate code). It's a bit restricted but might be sufficient for this rule. Check out https://github.com/realm/SwiftLint/blob/4078e41f1b1000a43cee065f3cd4d7160dd74911/Source/SwiftLintBuiltInRules/Rules/Style/PreferSelfInStaticReferencesRule.swift#L218 to get an idea of how this works.

Reply:
Thanks for the feedback. I was originally unsure how to avoid the duplication problem, but I didn't realize I didn't need an explicit rewriter and could use Visitor to handle corrections. 

I've tried to change this up a bit to use that pattern by taking advantage of some existing extensions on DeclModifierListSyntax. I'm just checking for the presence of a single access level modifier and substituting `private` in its place if it's higher than allowed for the scope of this rule. It doesn't necessarily solve for the `private(set) public` modifier situation, though.

Reply:
I wonder if `private(set) public` is a constellation, the rule should not complain about anyway. Are SwiftUI states allowed to be read by external clients? 🤔

Reply:
> So perhaps I could consider private(set) to be a non-triggering example...

Sounds reasonable. I'd support that.

Reply:
It turns out that declaring the access level of the property's setter as private causes the property to be exposed in the memberwise initializer. 

The following code will build:
```
public struct SwiftUIView: View {
    @State private(set) public var isPlaying: Bool = false

    public var body: some View {
        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
        Toggle(isOn: $isPlaying) {
            Text("Playing")
        }
    }
}

#Preview {
    SwiftUIView(isPlaying: true)
}
```

Declaring the State property as private prevents the value from being set in a memberwise initializer entirely. Based on that observation, I think that this rule needs to use strict enforcement of the private access level on each state property.

Reply:
I see. Thanks for investigating. In that case, an explicit rewriter would be the way to go.